Framework/Geometry/src/Instrument/Goniometer.cpp

```diff
@@ -128,6 +124,15 @@ void Goniometer::pushAxis(std::string name, double axisx, double axisy,
     throw std::runtime_error(
         "Initialized from a rotation matrix, so no axes can be pushed.");
   } else {
+    if (!boost::math::isfinite(axisx) || !boost::math::isfinite(axisy) ||
+        !boost::math::isfinite(axisz) || !boost::math::isfinite(angle)) {
```

Comment:
To get this to compile on Linux, it seems you either have to add `#include <boost/math/special_functions/fpclassify.hpp>` or change `boost::math::isfinite` to `std:isfinite` - I think the later is preferable, since we don't really need boost for this because the variables here are just `double`.